### PR TITLE
Rancher 186 - cypress tests pipeline #2

### PR DESCRIPTION
**Pull request by vkapylou**

### CODE REVIEW

File: pipelines/cypress.groovy [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            git branch: "${params.branch}",
+              url: 'https://github.com/folio-org/stripes-testing.git'
+          }
+        }
+        try {
+            docker.image('cypress/browsers:latest').inside('-u 0:0 --entrypoint=') {
+              dir('stripes-testing'){
+                stage('Execute cypress tests') {
+                    sh """
+                    id -u
```
Comment (OHaimanov) [inline pipelines/cypress.groovy:56]: It is unnecessary used just for permissions check

File: pipelines/cypress.groovy [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                    id -g
```
Comment (OHaimanov) [inline pipelines/cypress.groovy:57]: Same to previous

File: pipelines/cypress.groovy [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                    export CYPRESS_OKAPI_HOST=${params.OKAPI_HOST}
+                    export CYPRESS_OKAPI_TENANT=${params.OKAPI_TENANT}
+                    export CYPRESS_diku_login=${params.diku_login}
+                    export CYPRESS_diku_password=${params.diku_password}
+                    ls -la cypress/
+                    npx cypress run --headless ${params.run_param} || true
+                    chown -R 497:495 *
+                    find / -name allure-results
+                    ls -la /var/lib/jenkins/workspace/scratch_environment/folio-api-tests-cypress-v2.0/allure-results
+                    ls -la /var/lib/jenkins/workspace/scratch_environment/folio-api-tests-cypress-v2.0/target/allure-results
```
Comment (OHaimanov) [inline pipelines/cypress.groovy:72]: find and ls unnecessary used just for debug

File: pipelines/cypress.groovy [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                    yarn add @interactors/html --dev
+                    yarn add @interactors/html @interactors/with-cypress --dev
+                    export CYPRESS_BASE_URL=${params.baseUrl}
+                    export CYPRESS_OKAPI_HOST=${params.OKAPI_HOST}
+                    export CYPRESS_OKAPI_TENANT=${params.OKAPI_TENANT}
+                    export CYPRESS_diku_login=${params.diku_login}
+                    export CYPRESS_diku_password=${params.diku_password}
+                    ls -la cypress/
+                    npx cypress run --headless ${params.run_param} || true
+                    chown -R 497:495 *
```
Comment (OHaimanov) [inline pipelines/cypress.groovy:69]: It will be good to get this uid and gid with variables

File: pipelines/cypress.groovy [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                def allure_home = tool name: '2.17.2', type: 'allure'
+                sh "${allure_home}/bin/allure generate --clean "
+              }
+            }
+            stage('allure-results') {
+              dir('stripes-testing'){
+                sh '''
+                        id -u
+                        id -g
+                    '''
```
Comment (OHaimanov) [inline pipelines/cypress.groovy:91]: This sh step used for debug. Could be deleted

**Review by OHaimanov — CHANGES_REQUESTED** [review]

File: pipelines/cypress.groovy [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+              }
+            }
+        }
+    }
+    stage('Artifacts') {
+      dir('stripes-testing'){
+        archiveArtifacts artifacts: 'allure-results/*'
+      }
+    }
+}
```
Comment (OHaimanov) [inline pipelines/cypress.groovy:109]: Newline missed

File: pipelines/cypress.groovy [hunk 7]
```diff
@@ -0,0 +1,109 @@
+#!groovy
```
Comment (OHaimanov) [inline pipelines/cypress.groovy:1]: Please do squash to have only one commit

**Review by OHaimanov — APPROVED** [review]

File: pipelines/cypress.groovy [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+              export CYPRESS_diku_login=${params.diku_login}
+              export CYPRESS_diku_password=${params.diku_password}
+              npx cypress run --headless ${params.run_param} || true
+              chown -R $currentUID:$currentGID *
+            """
+          }
+        }
+      }
+    } catch (Exception err) {
+        println err
```
Comment (VitalyDemchenkoEpam) [inline pipelines/cypress.groovy:76]: for by understanding after adding "|| true" in 68 line try/catch is not required anymore?

File: pipelines/cypress.groovy [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    } finally {
+        stage('generate allure-report') {
+          dir('stripes-testing'){
+            def allure_home = tool name: '2.17.2', type: 'allure'
```
Comment (VitalyDemchenkoEpam) [inline pipelines/cypress.groovy:80]: Hardcoded version, please move to constants

File: pipelines/cypress.groovy [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+            sh "${allure_home}/bin/allure generate --clean "
+          }
+        }
+        stage('allure-results') {
+          dir('stripes-testing'){
+            allure([
+              includeProperties: false,
+              jdk              : '',
+              commandline      : '2.17.2',
```
Comment (VitalyDemchenkoEpam) [inline pipelines/cypress.groovy:89]: hardcoded version

**Pull request merged**